quantized_mesh_encoder/normals.py

```diff
@@ -56,13 +60,12 @@ def oct_encode(vec):
     l1_norm = np.linalg.norm(vec, ord=1, axis=1)
     result = vec[:, 0:2] / l1_norm[:, np.newaxis]
 
-    # TODO: Is 3rd position ever negative?
-    # Might depend on triangle winding order...
-    # if vec[2] < 0.0:
-    #     x = res[0]
-    #     y = res[1]
-    #     res[0] = (1.0 - abs(y)) * signNotZero(x)
-    #     res[1] = (1.0 - abs(x)) * signNotZero(y)
+    for i, res in enumerate(result):
+        if vec[i, 2] < 0.0:
+            x = res[0]
+            y = res[1]
+            res[0] = (1.0 - abs(y)) * signNotZero(x)
+            res[1] = (1.0 - abs(x)) * signNotZero(y)
```

Comment:
This will be incredibly slow because it's a Python for loop outside of C. It either needs to be numpy vectorized or put in Cython.

I think you should be able to do
```py
negative = vec[:, 2] < 0.0
x = result[:, 0]
y = result[:, 1]
result[:, 0] = np.where(negative, (1 - np.abs(y)) * sign_not_zero(x), result[:, 0])
result[:, 1] = np.where(negative, (1 - np.abs(x)) * sign_not_zero(y), result[:, 1])
```

You can/should benchmark it with `%%timeit`. This will probably be 50-100x faster than what you have above.

Reply:
Indeed with some simple benchmark tests it seems much much faster